client/network/test/src/lib.rs

```diff
@@ -789,6 +819,11 @@ pub trait TestNetFactory: Sized {
 		});
 	}
 
+	// TODO: Not ideal. Can we do better?
```

Comment:
What is not ideal?

Reply:
Thus far `client/network/test` did not require an executor but instead would defer execution to the user of the crate. Now, with this addition, it still depends on the user executing some components, but also spawns some components itself. This might be a surprise to some users.

I have not been able to come up with an alternative. Given that this is testing logic only, I don't think it is particularly important.

Reply:
> I have not been able to come up with an alternative. Given that this is testing logic only, I don't think it is particularly important.

I see. Seems fine to me, I would then just suggest to either remove the comment or make it more elaborate, so a reader is not left wondering what it is that is not ideal.

Reply:
I will remove it. There are still a couple of other outstanding `TODO` items which I need to address. I intend to address all of them before opening this pull request up for a full review. I should have documented this plan better in the pull request description.